versioned_docs/version-2.6/pages-for-subheaders/authentication-config.md

```diff
@@ -109,3 +109,23 @@ If you need to reconfigure or disable then re-enable a provider that had been pr
 is logged in to Rancher as an external user, not the local admin.
 
 :::
+
+## Disabling an auth provider
+
+When you disable an auth provider, Rancher deletes all resources associated with it, including secrets,
+global role bindings, cluster and role template bindings, and users who had never been local users in Rancher.
+
+As this operation may lead to a loss of many resources, you may want to add a saferuard on the provider.
+To ensure this cleanup process doesn't run when the auth provider is disabled, add a special annoation to the corresponding auth config.
+
+For example, to add a saferuard to the Azure AD provider, annotate the `azuread` authconfig object:
+
+`kubectl annotate --overwrite authconfig azuread management.cattle.io/auth-provider-cleanup='user-locked'`
+
+Rancher will not perform cleanup until you set the annotation's value to `unlocked`.
+
+### Running resource cleanup manually
+
+If you had configured an auth provider that is no longer enabled, Rancher might have still have some resources associated with the it in the local cluster.
```

Comment:
```suggestion
If you configure a disabled auth provider, Rancher might retain some resources associated with the provider in the local cluster.
```

Reply:
This one talks about a provider that used to be enabled, then got disabled and remains such. Not sure how to rephrase this...

Reply:
It sounds like Rancher is retaining old resources from when the provider was enabled, then? Maybe instead of trying to make small changes, we should try a new way to phrase.

"Rancher might retain resources from a previous auth provider configuration, if you  enable and disable the same provider, then attempt to configure the provider while it's still disabled."

Reply:
It's about a completely different provider. For example, a long time ago I had provider A. Then I disabled it. Instead, I enabled provider B and have been using it. Then I upgraded to a new version of Rancher. Now I can manually trigger resource cleanup for that disabled provider A, if I want.

Reply:
How about -- "Rancher might retain resources from a disabled auth provider configuration in the local cluster, even after you configure another auth provider. For example, if you used Provider A, then disabled them and started using Provider B, when you upgrade to a new version of Rancher you can manually trigger cleanup on resources configured by Provider A."